apps/hash-graph/lib/graph/src/api/rest.rs

```diff
@@ -194,6 +200,7 @@ where
     // The `/api-doc` endpoints are nested as we don't want any layers or handlers for the api-doc
     merged_routes
         .layer(Extension(dependencies.store))
+        .layer(Extension(dependencies.authorization_api))
```

Comment:
I've seen this a couple of times, but axum now recommends the use of a single `State` using `with_state` instead of an extension (https://docs.rs/axum/latest/axum/index.html#sharing-state-with-handlers) which also allows for substrates: https://docs.rs/axum/latest/axum/extract/struct.State.html#substates.

It might be worth to switch to a `State` object instead.

Reply:
I agree that we should switch but I'm not going to do this as a drive-by. There are a lot things which require improvements but if I would do this every time I touch the specific code it would take much longer. Thank you for pointing this out, though!

Reply:
Yea might have come across this way as a drive-by but was more like a note. Sorry!